src/plugins/data/public/search/session/sessions_client.ts

```diff
@@ -79,6 +79,12 @@ export class SessionsClient {
     });
   }
 
+  public extend(sessionId: string, keepAlive: string): Promise<SavedObjectsFindResponse> {
+    return this.http!.post(`/internal/session/${encodeURIComponent(sessionId)}/_extend`, {
+      body: JSON.stringify({ keepAlive }),
```

Comment:
I wouldn't call the parameter `keepAlive` as IMO it exposes some internals of the API

Reply:
Suggestions? Something like `timeToLive` or should we actually have the API be the `expires` value instead?

Reply:
I was thinking more about this today. The current UX will just extend the expiration by the `defaultExpiration` config value. The problem is that we need access to that value here in the `data` plugin (to extend each search request's `keep_alive`), but it's currently registered in the `data_enhanced` plugin config.

What I've done as a workaround is change the route (which is registered in `data_enhanced`) to calculate the `keep_alive` based on the `defaultExpiration`. Let me know what you think.

Reply:
Was just looking at [your PR](https://github.com/elastic/kibana/pull/89126). Since you're sending the `defaultExpiration` as the `extendBy` in the request, it negates the first problem I mentioned above. But then, the problem becomes that we'd need access to the `expires` attribute in the saved object, which currently is in a type that's only in `data_enhanced`. We may want to eventually move those types inside `data` rather than `data_enhanced`.